### ISSUE TITLE
Can't find any code for intermediate layer feature learning

### ISSUE THREAD

**Issue by iahmedmaher** [vicariousinc/science_rcn#20]
Does the published code not include the implementation of dictionary learning necessary for learning intermediate features? Why is the published code very limited and doesn't include everything in the paper?

**Comment by ayaz-amin** [comment]
If I recall correctly, intermediate feature learning does not exist on the two layer model. Correct me if I am wrong.

**Comment by iahmedmaher** [comment]
Yes, the channels, in this case, are known beforehand (the classes of the training set), but I expected that they would publish complete code except for the CAPTCHA part. Without intermediate feature learning, I can't try experiments on other more challenging dataset that require increasing the number of layers.

**Comment by ayaz-amin** [comment]
I am currently fixing up the code to run on Omniglot (I have cleaned it enough so that it works on Py3 without C++). I may try to get the intermediate feature learning to work (judging by the algorithm, it should not be that hard).

EDIT: On second thought, creating hierarchies beyond 2 layers is out of my scope.
